File: doc-examples/nitrite-example-java/src/main/java/example/VetoTimestampedRecordListener.java
package example;

import io.micronaut.data.event.EntityEventContext;
import io.micronaut.data.event.EntityEventListener;
import jakarta.inject.Singleton;

// tag::pre-persist-listener[]
@Singleton
public class VetoTimestampedRecordListener implements EntityEventListener<TimestampedRecord> {
    @Override
    public boolean prePersist(EntityEventContext<TimestampedRecord> context) {
        return !"veto-me".equals(context.getEntity().getName());
    }
}
// end::pre-persist-listener[]

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/VetoTimestampedRecordListener.kt
package example

import io.micronaut.data.event.EntityEventContext
import io.micronaut.data.event.EntityEventListener
import jakarta.inject.Singleton

// tag::pre-persist-listener[]
@Singleton
class VetoTimestampedRecordListener : EntityEventListener<TimestampedRecord> {
    override fun prePersist(context: EntityEventContext<TimestampedRecord>): Boolean {
        return context.entity.name != "veto-me"
    }
}
// end::pre-persist-listener[]
